src/QueueBatch/Impl/MessageBatch.cs

```diff
@@ -41,14 +43,16 @@ void IMessageBatch.MarkAllAsProcessed()
 
         public Task RetryAll(CancellationToken ct) => Complete(ct, Empty);
 
-        Task Complete(CancellationToken ct, HashSet<string> processed)
+        Task Complete(CancellationToken ct, IEnumerable<string> processed)
```

Comment:
@jeremy001181 Is there a reason why use `ConcurrentBag<string>` as the parameter type? This would enable to do not call `processed.ToList()` underneath, and do not allocate an additional list.

Reply:
No worries! I planned some time offline as well and wanted to ship this as promised. Therefore I took the liberty of adding one commit on top of yours. 1.1.0 has been released so I was able to keep the promise ;-)

Yeah, the concurrent bag does a linear search, so effectively, for n elements it might be O(n^2).

Reply:
@jeremy001181 ☝️